content/en/docs/advanced-tasks/catalog-update-formulary.md

```diff
@@ -0,0 +1,382 @@
+    packageName: testoperator
+    version: 1.1.0
+- type: olm.gvk
+  value:
+    group: api.testoperator.io
+    kind: Instance
+    version: v1alpha1
+relatedImages:
+- image: busybox
+  name: ""
+- image: quay.io/organization/testoperator:v1.1.0
+  name: ""
+schema: olm.bundle
+```
+
+## Formulae
+
+### Adding a new bundle to an existing channel
+#### semver
+Add a new `testoperatorv1.1.1` bundle pullspec to the `Candidate` channel archetype
+
+```bash
+yq eval '.candidate.bundles += [{"image" : "quay.io/organization/testoperator:v1.1.1"}]' semver.yaml
+```
+
+produces updated `Candidate` archetype contents:
+
+```yaml
+...
+candidate:
+  bundles:
+  - image: quay.io/organization/testoperator:v1.0.0
+  - image: quay.io/organization/testoperator:v1.0.1
+  - image: quay.io/organization/testoperator:v1.1.0
+  - image: quay.io/organization/testoperator:v1.1.1
+...
+```
+
+#### FBC
+Add a new `testoperator.v1.1.1` edge to an existing `candidate-v1.1` channel
+
+```bash
+yq eval 'select(.schema=="olm.channel" and .name == "candidate-v1.1").entries += [{"name" : "testoperator.v1.1.1"}]' fbc.yaml
+```
+
+produces updated `candidate-v1.1` channel:
+
+```yaml
+...
+entries:
+  - name: testoperator.v1.1.0
+    replaces: testoperator.v1.0.1
+    skips:
+      - testoperator.v1.0.0
+  - name: testoperator.v1.1.1
+name: candidate-v1.1
+package: testoperator
+schema: olm.channel
+...
+```
+
+
+### Adding a new 'replaces' link between two existing bundles
+
+#### FBC
+Adding a new `testoperator.v1.1.1` bundle version edge with a replaces link to its predecessor `testoperator.v1.1.0` version which already exists in the channel. 
+
+```bash
+yq eval 'select(.schema == "olm.channel" and .name == "candidate-v1.1").entries += [{"name" : "testoperator:v1.1.1", "replaces": "testoperator:v1.1.0"}]' fbc.yaml
+```
+produces updated `candidate-v1.1` channel:
+
+```yaml
+...
+entries:
+  - name: testoperator.v1.1.0
+    replaces: testoperator.v1.0.1
+    skips:
+      - testoperator.v1.0.0
+  - name: testoperator:v1.1.1
+    replaces: testoperator:v1.1.0
+name: candidate-v1.1
+package: testoperator
+schema: olm.channel
+...
+```
+
+### Removing a specific bundle version
+
+#### FBC
+Remove the upgrade edge from the example `candidate-v1.1` channel which refers to bundle version `testoperator.v1.1.0`. 
+
+```bash
+yq eval 'del(select(.schema == "olm.channel" and .name == "candidate-v1.1" ).entries[]| select(.name == "testoperator.v1.1.0"))' fbc.yaml
+```
+
+produces updated `candidate-v1.1` channel:
+
+```yaml
+...
+entries: []
+name: candidate-v1.1
+package: testoperator
+schema: olm.channel
+...
+```
+Please note that removing the only edge for a channel as in this example will yield an explicitly empty array.  This will produce an error in `opm validate`. 
+
+### Substituting a bundle version in the upgrade graph
+
+#### semver
+For all channels, replace instances of `quay.io/organization/testoperator:v1.1.0` with `quay.io/organization/testoperator:v1.1.0-CVE`
```

Comment:
I don't think removal of one bundle and addition of another works if the original catalog has already been shipped to users.

Reason being that someone may have already installed `quay.io/organization/testoperator:v1.1.0`. If it is no longer in the graph, it means that nothing knows to upgrade from it.

Reply:
This case is specifically for the freshmaker use-case, but it could just as easily be used by someone who spun a custom catalog and had to re-issue the bundle for some reason as an 'oops'.  
It's probably a good idea to capture the dangers of willy-nilly removing a version from a graph, but I was trying to not get too encumbered with "when would this case be useful" commentary.

Reply:
> This case is specifically for the freshmaker use-case

Yeah, that's my concern actually. In this use case, it is actually _really_ important that any users on the existing "bad" version have an off-ramp. Personally, I don't think its _ever_ a good idea to strand bundles by removing them without proving upgrade edges out to other versions that remain. And if I were a catalog maintainer, I'd want to ban it.

I think the only way to do this correctly with the semver template (as it is currently implemented), is to simply add another bundle that is either the next patch version or a higher pre-release version. The result would be that the "bad" image is skipped by the new version and nothing upgrades to it.